rfcs/convergence/props-consistency-check.md

```diff
@@ -0,0 +1,391 @@
+# Consistency Check
+
+The following components and their props were reviewed as of 9/7/2021 for consistency.
+
+Accordian, AccordianHeader, AccordianItem, AccordianPanel
+Avatar
+Badge, CounterBadge, PresenceBadge,
+Button, CompoundButton, MenuButton, ToggleButton,
+Divider,
+Image,
+Label,
+Link,Menu, MenuDivider, MenuGroup, MenuGroupHeader,MenuItem, MenuItemCheckbox, MenuItemRadio, MenuList, MenuPopover, MenuTrigger
+Popover, PopoverSurface, PopoverTrigger
+Portal
+
+## Nit: Explicit component type declaration
+
+> Some components declare their type explicity, others do not.
+
+```tsx
+export const Accordion: React.FunctionComponent<AccordionProps & React.RefAttributes<HTMLElement>> = React.forwardRef<
+  HTMLElement,
+  AccordionProps
+>((props, ref) =>
+```
+
+```tsx
+export const Avatar = React.forwardRef((props: AvatarProps, ref: React.Ref<HTMLElement>) =>
+```
+
+## Props declared as interfaces vs types
+
+> Some props are defined as interfaces and extended. Others are defined as types and intersected. This isn't always consistent within the same component.
+
+Note: I won't list out every component here. We can scrub once we decide the right approach.
+
+```ts
+export interface AccordionProps extends ComponentProps<AccordionSlots>, Partial<AccordionCommons>
+
+export type AccordionSlots
+export interface AccordionCommons
+```
+
+```ts
+export interface AccordionHeaderProps extends ComponentProps<AccordionHeaderSlots>, Partial<AccordionHeaderCommons>
+
+export type AccordionHeaderSlots
+export interface AccordionHeaderCommons
+```
+
+## Props extending ComponentProps vs. ComponentPropsCompat
```

Comment:
from @ecraig12345 in HackMD:
Pretty sure this is planned to be addressed before beta with the simplified prop merging issues

Reply:
Agree